Atomic/discovery.py

```diff
@@ -21,17 +21,26 @@ def __init__(self, registry=None, repo=None, image=None, tag=None, digest=None,
         self.digest = digest
         self.orig_input = orig_input
         self._remote_inspect = None
-        self.fqdn = None
+        self._fqdn = None
 
         if self.debug:
             util.output_json(self.registries)
 
+    @property
+    def fqdn(self):
+        if not self._fqdn:
+            self.fqdn = self.assemble_fqdn(include_tag=True) if self.registry else self.find_image_on_registry()
```

Comment:
Maybe I'm misunderstanding, but should this be "self._fqdn = self.assemble_fqdn" here now?